### DISCUSSION TITLE
Having trouble with nested-lists and tutorial 21

### DISCUSSION THREAD

**Discussion by sabracrolleton** [rabbibotton/clog#126] [Q&A] [answered]
Consider a hierarchy tree built of lists and sublists that I will not know until runtime. I want to start out collapsed and choose which subtrees to expand, then click on an actual final item to get an alert box with info about the item. In tutorials 19 and 20, it is very clear and easy to create a subtree. In tutorial 19 You can just drop a 

    (create-unordered-list item)

in the middle and now the previous item becomes the top of a new subtree. 

    (let* ((list-top   (create-unordered-list body))
	     (item       (create-list-item list-top :content "Top of tree"))
	     (list-b     (create-unordered-list item))
	     (item       (create-list-item list-b :content "Item 1"))
	     (item       (create-list-item list-b :content "Middle Drop"))
         (list-c     (create-unordered-list item))
	     (item       (create-list-item list-c :content "Item 3"))
	     (item       (create-list-item list-c :content "Item 4")))

It starts out nicely collapsed and I can pick and choose what to expand and what to collapse. It is clear what is the beginning of another tree and what is a final leaf of a branch.

I am struggling to do the same in tutorial 21. 

Any pointers on how to do that?
Specifically, consider tutorial 21 with the following on-new-window function

    (defun on-new-window (body)
        (setf (title (html-document body)) "Tutorial 21")
      (let* ((drop-list  (clog-drop-list:create-drop-list body :content "Top of tree"))
	         (item (create-list-item (clog-drop-list:drop-root drop-list) :content "Item 1"))
    	 (item (create-list-item (clog-drop-list:drop-root drop-list) :content "Item 2"))
             (drop-list-a  (clog-drop-list:create-drop-list item :content "Middle of tree"))
	         (item (create-list-item (clog-drop-list:drop-root drop-list-a) :content "Item 3"))
	         (item (create-list-item (clog-drop-list:drop-root drop-list-a) :content "Item 4"))
             (drop-list-b  (clog-drop-list:create-drop-list item :content "Lower Part of tree"))
	         (item (create-list-item (clog-drop-list:drop-root drop-list-b) :content "Item 5"))
	         (item (create-list-item (clog-drop-list:drop-root drop-list-b) :content "Item 6")))
        (declare (ignore item))))

Ignoring the fact that I do not actually know the number of items and subtrees until runtime, tutorial 21 with the above tree opens fully expanded. 

If I then click on the bottom subtree, everything collapses. 

Clicking on the top of tree at that point returns Item 1 and Item 2 and Middle of tree.

If I click on Middle of tree trying to get its sublist, everything collapses back to Top of tree

Clicking on Top of Tree again now expands down to Lower Part of Tree (without sublist) 

Clicking on Lower Part of Tree collapses everything again; this time clicking on Top of tree returns the first two items and Middle of Tree. Clicking anywhere collapses everything again, but this time clicking on Top of tree expands the entire tree.

Basically I am misunderstanding how the hide/not hide is passed up and down.

**Comment by rabbibotton** [comment, marked as answer]
Since the example was emphasizing how to write a plug-in I never tested it with additional drop downs. I have corrected the example. The correction is I used on-mouse-down instead of on-click and I set cancel-event to true. This prevents "event bubble" so that the same even after processing bubbles up to the outer containing element.

I also added more to the example so you can see how this works and also how to control drops at run time.

Sorry for the delay in answering.

**Reply by rabbibotton** [reply]
Make sure you are using the GitHub version and do a - git pull
to get the new tutorial-21
https://github.com/rabbibotton/clog/blob/main/tutorial/21-tutorial.lisp
